### PR TITLE
server: move mock relay into mock package

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
## 📝 Summary
This PR moves the mock relay into its own package and moves some shared dependencies into their own packages in order to simplify the server package.

## ⛱ Motivation and Context
In order to simplify the architecture of mev-boost, it makes sense in my opinion to move certain functionalities into their own package. This makes it easier to reason about components and abstractions and its easier to add some more tests and refactors in the future.
This is just an initial PR to move the files, it changes no behavior. I will send some follow-up PRs to clean up the server code and the mock relay

### CODE REVIEW

**Review by jtraglia — CHANGES_REQUESTED** [review]
This looks reasonable to me. Thanks! One thing I'd like to be fixed first though.

File: server/mock/mock_relay.go [hunk 1]
```diff
@@ -153,9 +155,9 @@ func (m *mockRelay) handleRegisterValidator(w http.ResponseWriter, req *http.Req
 }
 
 // defaultHandleRegisterValidator returns the default handler for handleRegisterValidator
-func (m *mockRelay) defaultHandleRegisterValidator(w http.ResponseWriter, req *http.Request) {
+func (m *Relay) defaultHandleRegisterValidator(w http.ResponseWriter, req *http.Request) {
 	payload := []builderApiV1.SignedValidatorRegistration{}
-	if err := DecodeJSON(req.Body, &payload); err != nil {
+	if err := json.NewDecoder(req.Body).Decode(&payload); err != nil {
```
Comment (jtraglia) [inline server/mock/mock_relay.go]: This has differing functionality. This will allow unknown fields.

Comment (MariusVanDerWijden) [reply server/mock/mock_relay.go]: yep, fixed

**Review by jtraglia — APPROVED** [review]
LGTM 🥇

**Review by metachris — APPROVED** [review]
thanks!

**Pull request merged**